src/System.Net.WebSockets.Client/tests/ClientWebSocketTest.cs

```diff
@@ -465,6 +509,89 @@ public static IEnumerable<object[]> UnavailableWebSocketServers
         }
 
         [ConditionalTheory("WebSocketsSupported"), MemberData("EchoServers")]
+        public async Task CloseOutputAsync_ClientInitiated_CanReceive_CanClose(Uri server)
+        {
+            string message = "Hello WebSockets!";
+
+            using (ClientWebSocket cws = await WebSocketHelper.GetConnectedWebSocket(server, TimeOutMilliseconds, _output))
+            {
+                var cts = new CancellationTokenSource(TimeOutMilliseconds);
+
+                var closeStatus = WebSocketCloseStatus.InvalidPayloadData;
+                string closeDescription = "CloseOutputAsync_Client_InvalidPayloadData";
+
+                await cws.SendAsync(WebSocketData.GetBufferFromText(message), WebSocketMessageType.Text, true, cts.Token);
+                // Need a short delay as per WebSocket rfc6455 section 5.5.1 there isn't a requirement to receive any
+                // data fragments after a close has been sent. The delay allows the received data fragment to be
+                // available before calling close. The WinRT MessageWebSocket implementation doesn't allow receiving
+                // after a call to Close.
```

Comment:
These particular tests being ported have been very reliable so far. The test server endpoints for the GitHub CoreFx repo run in the Azure Cloud using fault tolerant multiple instance web roles. So, it is unlikely to have any downtime. That said, we've seen occasional Azure hiccups and that would affect all tests (http, websocket, etc.).

We'll keep an eye on these tests and make any adjustments needed.